src/services/helper/mentors.js

```diff
@@ -76,28 +77,42 @@ module.exports = class MentorsHelper {
 	 * @returns {JSON} - profile details
 	 */
 	static async profile(id) {
-		const mentorsDetails = await userProfile.details('', id)
-		if (mentorsDetails.data.result.isAMentor) {
-			const _id = mentorsDetails.data.result._id
-			const filterSessionAttended = { userId: _id, isSessionAttended: true }
-			const totalSessionsAttended = await sessionAttendees.countAllSessionAttendees(filterSessionAttended)
-			const filterSessionHosted = { userId: _id, status: 'completed', isStarted: true }
-			const totalSessionHosted = await sessionsData.findSessionHosted(filterSessionHosted)
-			return common.successResponse({
-				statusCode: httpStatusCode.ok,
-				message: 'PROFILE_FTECHED_SUCCESSFULLY',
-				result: {
-					sessionsAttended: totalSessionsAttended,
-					sessionsHosted: totalSessionHosted,
-					...mentorsDetails.data.result,
-				},
-			})
-		} else {
-			return common.failureResponse({
-				statusCode: httpStatusCode.bad_request,
-				message: 'MENTORS_NOT_FOUND',
-				responseCode: 'CLIENT_ERROR',
-			})
+		try {
+			let mentorsDetails = {}
+			if (await RedisHelper.getKey(id)) {
```

Comment:
@ankitpws for checking the key you are getting cache data and again calling in the set line to send the, change this.

Reply:
@ankitpws profile caching should be done in user service or here ?